e2e/testcases/preserve_fields_test.go

```diff
@@ -100,19 +100,17 @@ func TestPreserveGeneratedServiceFields(t *testing.T) {
 	generatedNodePort := gotService.Spec.Ports[0].NodePort
 	generatedClusterIP := gotService.Spec.ClusterIP
 
-	// 5 seconds is more than enough time for this to happen.
-	_, err = nomostest.Retry(5*time.Second, func() error {
-		// This can only return nil if the NodePort/ClusterIP was updated.
-		// Potentially flaky check since other things can cause NodePort/ClusterIP
-		// to change; copied from bats.
-		return nt.Validate(serviceName, ns, &corev1.Service{},
-			hasDifferentNodePortOrClusterIP(generatedNodePort, generatedClusterIP))
-	})
+	// This can only return nil if the NodePort/ClusterIP was updated.
+	// Potentially flaky check since other things can cause NodePort/ClusterIP
+	// to change; copied from bats.
+	err = nomostest.WatchObject(nt, kinds.Service(), serviceName, ns,
+		[]nomostest.Predicate{hasDifferentNodePortOrClusterIP(generatedNodePort, generatedClusterIP)},
+		nomostest.WatchTimeout(30*time.Second))
```

Comment:
hmm -> why the bump from 5 to 30 seconds?

Reply:
Any change that should happen in 5s could be much longer if the apiserver is congested or the local test machine is slow.